framework/source/class/qx/ui/basic/Label.js

```diff
@@ -535,7 +535,12 @@ qx.Class.define("qx.ui.basic.Label",
     _applyValue : function(value, old)
     {
       // Sync with content element
-      this.getContentElement().setValue(value);
+      if (value && qx.core.Environment.get("qx.dynlocale") && value.translate) {
```

Comment:
so if we are talking performance, I would call

````javascript
if (value && value.translate && qx.core.Environment.get("qx.dynlocale")) {
````
as this would make the most expensive operation be called only if both the cheep operations are ok.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Short-Circuit_Evaluation